desmod/component.py

```diff
@@ -275,6 +275,23 @@ def elab_hook(self):
         """
         pass
 
+    def post_fail(self):
+        """Recursively run post-fail hooks."""
+        for child in self._children:
+            child.post_fail()
+        self.post_sim_fail_hook()
+
+    def post_sim_fail_hook(self):
```

Comment:
`post_sim_fail_hook()` --> `exception_hook()`
`post_fail()` --> `on_exception()`

But this is naming is a secondary issue. The big issue is whether this paradigm is actually viable/appropriate.